tests/tools/oesign-engine/test-enclave/oesign-test.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+#!/usr/bin/env python3
+#  Run tests of oesign.
+#  We need this wrapper to return the correct return value from the tests, since
+#  several invocations of oesign are supposed to fail by design.
```

Comment:
Not quite sure why this is necessary? You're already parameterizing the inputs to each of the tests, except that you're also choosing to hardcode a value in each one as well (e.g. `oesign_detects_failed_engine_id` when `engine_id` is already an input parameter that's ignored by the test)

It seems like you could create a single function that consolidates all of the variants and add a input parameter for expected success/failure, then it'd be fully parametrized and it wouldn't require parsing out separate definitions.

Reply:
The standardised parameter list allows me to put those functions in a dict.  
If I modified the functions to take specific parameters or to set variables or whatever they would actually result in less declarative, longer code.